main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1551,6 +1576,17 @@ object Typer extends Phase[ResolvedAst.Root, TypedAst.Root] {
         val eff = e.eff
         TypedAst.Expression.Spawn(e, tpe, eff, loc)
 
+      case ResolvedAst.Expression.Lazy(exp, loc) =>
+        val e = visitExp(exp, subst0)
+        val tpe = Type.mkLazy(e.tpe)
```

Comment:
I think we need to use the local mkApply for kind safety (?)

Reply:
Maybe, but not here, right? This is where the AST is rebuilt.

Reply:
Right, my bad. Looks like that's already taken care of in the resolver